nautobot/docs/development/apps/migration/migrating-an-app-from-v1.md

```diff
@@ -0,0 +1,191 @@
+# Migration Guide to Upgrade an App from V1 to V2
+
+## Model Updates
+
+### Core
+
+#### Replace the Usage of Slugs with Composite Keys
+
+Slugs were used to identify unique objects in the database for various models in Nautobot v1.x and they are now replaced by Composite Keys. The `slug` field can be safely deleted as long as your models are derived from `BaseModel` that automatically supports the following [natural key](https://docs.djangoproject.com/en/3.2/topics/serialization#natural-keys) APIs. For a more comprehensive guideline on how Natural Keys/Composite Keys in Nautobot v2.0 works, please go to the [Natural Key documentation](../../core/natural-keys.md).
```

Comment:
Probably need to call out any changes to plugin URL patterns that may result from removing slugs?